### ISSUE TITLE
Changes to clippy have broken CI on Tokio

### ISSUE THREAD

**Issue by Darksonn** [taiki-e/pin-project-lite#46]
The Tokio master branch failed with the following error:
```
error: unknown clippy lint: clippy::ref_option_ref
   --> tokio/src/task/local.rs:150:1
    |
150 | / pin_project! {
151 | |     #[derive(Debug)]
152 | |     struct RunUntil<'a, F> {
153 | |         local_set: &'a LocalSet,
...   |
156 | |     }
157 | | }
    | |_^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unknown_clippy_lints
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by taiki-e** [comment]
Thanks for the report! This is definitely clippy's bug (`unknown_clippy_lints` shouldn't warn of code generated by external macros), but I will fix it soon on the pin-project-lite side. (the same issue as https://github.com/taiki-e/pin-project/pull/303)

**Comment by taiki-e** [comment]
Published 0.2.3 which fixes this issue.

**Comment by Darksonn** [comment]
Thanks!
